bokehjs/src/coffee/models/tools/draw/edit_tool.ts

```diff
@@ -0,0 +1,52 @@
+import {SelectToolView} from "../gestures/select_tool"
+
+export interface BkEv {
+  bokeh: {
+    sx: number
+    sy: number
+  }
+  srcEvent: {
+    shiftKey?: boolean
+  }
+  keyCode: number
+  timeStamp: number
+}
+
+export abstract class EditToolView extends SelectToolView {
+
+  // this is executed when the pan/drag event starts
+  _map_drag(sx: number[], sy: number[]): [number, number] {
+    const frame = this.plot_model.frame;
+
+    if (!frame.bbox.contains(sx, sy)) {
+      return
```

Comment:
The return type of this method is `[number, number]`, but you are returning (implicitly) `undefined`, which would imply `[number, number] | undefined`. However, the more idiomatic approach is to return `null`, to indicate return of a proper value, and so the return type should be changed to `[number, number] | null`.

Reply:
Thanks, that makes sense.